src/java/org/apache/cassandra/service/accord/interop/AccordInteropExecution.java

```diff
@@ -290,6 +291,66 @@ private AsyncChain<Data> readChains()
                 }));
             });
         });
+        return results;
+    }
+
+    private List<AsyncChain<Data>> rangeReadChains(long nowInSeconds, Dispatcher.RequestTime requestTime)
+    {
+        TxnRangeRead read = (TxnRangeRead) txn.read();
+        Seekables<?, ?> keys = txn.read().keys();
+        List<AsyncChain<Data>> results = new ArrayList<>();
+        keys.forEach(key -> {
+            TokenRange range = (TokenRange)key;
+            PartitionRangeReadCommand command = read.commandForSubrange(range, nowInSeconds);
+
+            // This should only rarely occur when coordinators start a transaction in a migrating range
+            // because they haven't yet updated their cluster metadata.
+            // It would be harmless to do the read, because it will be rejected in `TxnQuery` anyways,
+            // but it's faster to skip the read
+            // TODO (required): To make migration work we need to validate that the range is all on Accord
+            // if any part isn't we should reject the read
```

Comment:
so if part of the range isn't on accord, then we don't start the migration and sequence the read after the migration?  Rejecting the read would cause the user to see an error, so we would break SLO?

Reply:
We can't forcibly sequence things running on the correct system because Accord might run the transaction in a later epoch from when it was submitted and we only find out when we go to execute it.

Here we need to validate like keys do that every key being read is readable by Accord.

Users don't see the error because it's automatically retried again on the correct system.